### ISSUE TITLE
Crop after 90/270deg rotation results in "Error: extract_area: bad extract area"

### ISSUE THREAD

**Issue by asilvas** [lovell/sharp#296]
I've verified the resulting image dimensions are correct (width & height are effectively flipped), so it appears the crop operation is not taking into account the new image dimensions.

```
Error: extract_area: bad extract area

    at Error (native)
```

**Comment by lovell** [comment]
Hi Aaron, there is [existing logic](https://github.com/lovell/sharp/blob/master/src/pipeline.cc#L285-L293) and [test cases](https://github.com/lovell/sharp/blob/master/test/unit/extract.js#L98-L120) to deal with this situation.

The ordering of `extract` and `resize` is significant. It's possible to `extract` two different areas in the same pipeline, before and after `resize`. Are you able to share some sample code?

**Comment by asilvas** [comment]
I don't have code I can copy/paste, but basically this is happening:

```
sharp(file200by100).rotate(90).crop(0,0,100,200);
```

If I use the minimum dimension it'll work, but for whatever reason the error is being thrown when it shouldn't be.

**Comment by lovell** [comment]
I've been able to reproduce this problem with the following:

```javascript
sharp('test/fixtures/2569067123_aca715a2ee_o.jpg')
  .resize(200, 100)
  .toBuffer()
  .then(function(data) {
    return sharp(data)
      .rotate(90)
      .extract(0, 0, 100, 200)
      .toBuffer();
  }).catch(function(err) {
    console.log(err);
  });
```
It occurs when rotating and extracting without (or before) a resize in the same pipeline.

Should be fairly easy to fix and add tests for this combination. Thanks for reporting!

**Comment by asilvas** [comment]
Awesome, thanks!

**Comment by lovell** [comment]
Commit 58d9e0f should fix this. Are you able to retest using the `look` branch?

```sh
npm install lovell/sharp#look
```

(This is the work-in-progress branch for what will become the v0.12.0 release.)

**Comment by asilvas** [comment]
Verified working.

**Comment by lovell** [comment]
Great, thanks for confirming Aaron. This will be included as part of v0.12.0.

If you're able to share details publicly, it's always great to learn about what people are using this module for in the survey at #35.

**Comment by lovell** [comment]
The `look` branch has now been merged to the `master` branch pending release. Thanks again for reporting this.
